packages/sitecore-jss-nextjs/src/components/NextImage.test.tsx

```diff
@@ -0,0 +1,323 @@
+          contentSecurityPolicy: 'test',
+        },
+        src: 'https://cm.jss.localhost/assets/img/test0.png?mw=100',
+        width: 100,
+      };
+      const result = loader(params);
+      expect(result).to.be.a('string');
+      expect(result).to.equal('https://cm.jss.localhost/assets/img/test0.png?mw=100');
+    });
+
+    it('should throw an error if path is not configured', () => {
+      const params: ImageLoaderProps = {
+        config: {
+          deviceSizes: [],
+          imageSizes: [],
+          path: undefined,
+          allSizes: [],
+          loader: 'default',
+          domains: [],
+          disableStaticImages: false,
+          minimumCacheTTL: 1,
+          formats: [],
+          dangerouslyAllowSVG: false,
+          contentSecurityPolicy: 'test',
+        },
+        src: '/assets/img/test0.png?mw=100',
+        width: 100,
+      };
+      expect(() => loader(params)).to.throw(
+        'Failed to load image. Please make sure images path is configured correctly in next.config.js'
+      );
+    });
+  });
+
+  describe('error cases', () => {
+    const src = '/assets/img/test0.png';
+    it('should throw an error if src is present', () => {
+      expect(() => mount(<NextImage src={src} />)).to.throw(
+        'Detected src prop. If you wish to use src, use next/image directly.'
+      );
+    });
+  });
+
+  describe('with direct image object, no value/editable', () => {
+    const props = {
+      field: {
+        src: '/assets/img/test0.png',
+      },
+      width: 8,
+      height: 10,
+    };
+
+    const rendered = mount(<NextImage loader={mockLoader} {...props} />).find('Image');
+    it('should render image with url', () => {
+      rendered.render();
+      expect(rendered).to.have.lengthOf(1);
+      expect(rendered.prop('src')).to.equal(props.field.src);
+      expect(rendered.prop('width')).to.equal(props.width);
+      expect(rendered.prop('height')).to.equal(props.height);
+
+      expect(mockLoader.called).to.be.true;
+      expect(mockLoader).to.have.been.calledWith(loaderParams);
+    });
+  });
+});
+
+describe('With loader function passed by the user', () => {
+  const userCustomLoader = () => new URL('https://cm.jss.localhost').href;
+  const userMockLoader = (stub(userCustomLoader) as unknown) as ImageLoader;
+  const props = {
+    field: {
+      src: '/assets/img/test0.png',
+    },
+    width: 8,
+    height: 10,
+    loader: userMockLoader,
+  };
+
+  const rendered = mount(<NextImage {...props} />).find('Image');
+
+  it('should render image with url', () => {
+    expect(rendered).to.have.lengthOf(1);
+    expect(rendered.prop('src')).to.equal(props.field.src);
+    expect(rendered.prop('width')).to.equal(props.width);
+    expect(rendered.prop('height')).to.equal(props.height);
+    expect(rendered.prop('loader')).to.equal(props.loader);
+    expect(mockLoader.called).to.be.true;
+    expect(mockLoader).to.have.been.calledWith(loaderParams);
```

Comment:
This is using the outer-scoped `mockLoader` - should be using `userMockLoader`.  Guessing this is working only b/c other tests are running previously. We should add an `afterEach` and call `mockLoader.reset()` to make sure there is no history leaking between tests.